### ISSUE TITLE
Test results dependent on existance of `log()`

### ISSUE THREAD

**Issue by nfurfaro** [FuelLabs/sway#4307]
TL;DR: adding or removing a simple log() is affecting the outcome of tests.
Details:
- the errors I'm seeing are `MemoryOverflow` (and the U256 type is involved...)
- toolchain is beta-3
- repo is https://github.com/FuelLabs/bridge-fungible-token
- branch is furnic/abi-split-2
- This doesn't happen on `master`. The changes in my PR are mainly refactoring/reorganizing modules as you can see in the diff for the PR here: https://github.com/FuelLabs/bridge-fungible-token/pull/57/files
 
**To reproduce:**
1. clone the repo linked above
2. check out the branch `furnic/abi-split-2`
3. set the toolchain with `fuelup default beta-3` 
4. in the root of the repo, run `forc build`
5. in the root of the repo, run `cargo test`

You should get a result like this:
![image](https://user-images.githubusercontent.com/17053076/225973732-c2535456-c41e-4d23-9160-7b0a17778e79.png)

6. Now, in the contract at `bridge-fungible-token/bridge-fungible-token/utils.sw`, add the following on line 93:
`log(111);`

7. Next, rebuild the project and rerun the tests. They should all pass now.

**Comment by nfurfaro** [comment]
Update: the beta-3 toolchain has been updated, the the issue still persists.
The current versions I'm using to test this behaviour are:
![image](https://user-images.githubusercontent.com/17053076/227561491-36dc1da7-b1e8-4ca2-9d82-ff028e1239ad.png)

**Comment by IGI-111** [comment]
From #6352:

> ### Problem
> The given test only passes if the logging of the result occurs (may not be limited to logging, could just be usage of the result in anyway), if the log call is commented out, the test fails
> 
> ### Steps
>  ```sway
> #[test(should_revert)]
> fn math_u64_pow_overflow_should_revert() {
>     let _result = 2_u64.pow(64);
>     log(_result);
> }
> ```
> forc test
